src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,205 @@
+        }
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean checkWhitespaceBeforeColonOfCase(DetailAST colonAst) {
+        final DetailAST previousSibling = colonAst.getPreviousSibling();
+        int offset = 0;
+        if (previousSibling.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {
+            offset = 1;
+        }
+        return colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + offset;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean checkWhitespaceBeforeColonOfDefault(DetailAST colonAst) {
+        final boolean horizontalWhitespace;
+        final DetailAST commentBlock = colonAst.getPreviousSibling();
+        if (commentBlock == null) {
+            final DetailAST literalDefault = colonAst.getParent();
+            horizontalWhitespace = colonAst.getColumnNo()
+                    != literalDefault.getColumnNo() + literalDefault.getText().length();
+        }
+        else {
+            horizontalWhitespace =
+                    colonAst.getColumnNo() != getLastColumnNumberOf(commentBlock) + 1;
+        }
+        return horizontalWhitespace;
+    }
+
+    /**
+     * Returns the last column number of an ast.
+     *
+     * @param ast DetailAST to check.
+     * @return ast's last column number.
+     */
+    private static int getLastColumnNumberOf(DetailAST ast) {
+        DetailAST lastChild = ast;
+        while (lastChild.hasChildren()) {
+            lastChild = lastChild.getLastChild();
+        }
+        return lastChild.getColumnNo() + lastChild.getText().length();
+    }
+
+    /**
+     * Checks if the colon is on same line as of case or default.
+     *
+     * @param colonAst DetailAST to check.
+     * @param isColonOfCase boolean parameter representing if the colon is after case.
+     * @param isColonOfDefault boolean parameter representing if the colon is after default.
+     * @return true, if colon case is in different line as of case or default.
+     */
+    private static boolean isColonOnDifferentLine(DetailAST colonAst, boolean isColonOfCase,
+                                      boolean isColonOfDefault) {
+        boolean sameLine = true;
+        if (isColonOfCase) {
+            final DetailAST previousSibling = colonAst.getPreviousSibling();
+            sameLine = TokenUtil.areOnSameLine(previousSibling, colonAst);
+        }
+        else if (isColonOfDefault) {
+            final DetailAST literalDefault = colonAst.getParent();
+            sameLine = TokenUtil.areOnSameLine(literalDefault, colonAst);
+        }
+        return !sameLine;
+    }
```

Comment:
Can we extract `previousSibling` and `literalDefault` to just have a `previousToken`? Then we can just return `!TokenUtil.areOnSameLine(previousToken, colonAst);` I would also just pass in the parent token that you have already declared in `visitToken` and check for it's type in the `if`/ `else if` conditions, to eliminate the boolean parameters.

Something like:
```java
    private static boolean isColonOnDifferentLine(DetailAST colonAst, DetailAst parent) {
        final DetailAst previousToken;
        if (parent.getType() == TokenTypes.LITERAL_CASE) {
            previousToken = colonAst.getPreviousSibling();
        }
        else if (parent.getType() == TokenTypes.LITERAL_DEFAULT) {
            previousToken = colonAst.getParent();
        }
        return !TokenUtil.areOnSameLine(previousToken, colonAst);
     }

```

Reply:
Done.

Reply:
Resolved.